scripts/quests/bastok/Return_to_the_Depths.lua

```diff
@@ -0,0 +1,256 @@
+                    player:delKeyItem(xi.ki.PROVIDENCE_POT)
+                    player:delKeyItem(xi.ki.LETTER_FROM_MAGRIFFON)
+                    player:delKeyItem(xi.ki.LETTER_FROM_MUCKVIX)
+                    player:messageSpecial(zones[player:getZoneID()].text.KEYITEM_LOST, xi.ki.PROVIDENCE_POT)
+                    player:messageSpecial(zones[player:getZoneID()].text.KEYITEM_LOST, xi.ki.LETTER_FROM_MAGRIFFON)
+                    player:messageSpecial(zones[player:getZoneID()].text.KEYITEM_LOST, xi.ki.LETTER_FROM_MUCKVIX)
+                    npcUtil.giveKeyItem(player, xi.ki.PUNGENT_PROVIDENCE_POT)
+                    quest:setVar(player, 'prog', 5)
+                end,
+            },
+        },
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Magriffon'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 2 then
+                        return quest:progressEvent(299, 66, 630, 10000, 0, 65994751, 7080303, 4095, 4)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { {"gil", 10000} } ) and
+                    quest:getVar(player, 'prog') == 3 then
+                        return quest:progressEvent(301, 75, 630, 10000, 0, 65994751, 7080303, 4095, 4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [299] = function(player, csid, option, npc)
+                    return quest:setVar(player, 'prog', 3)
+                end,
+
+                [301] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, {xi.ki.PROVIDENCE_POT, xi.ki.LETTER_FROM_MAGRIFFON})
+                    quest:setVar(player, 'prog', 4)
+                end,
+            },
+        },
+
+        [xi.zone.MINE_SHAFT_2716] =
+        {
+            ['_0d0'] = -- Mine Shaft BCNM Entrance
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 8 then
+                        return quest:progressEvent(5, 1, 5, 0, 24, 1661, 0, 0, 65536)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [5] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 9)
+                end,
+            },
+        },
+    },
+
+    -- Completed
+    {
+        [xi.zone.PORT_BASTOK] =
+        {
+            ['Ravorara'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.PUNGENT_PROVIDENCE_POT) then
+                        return quest:progressEvent(313, 653325294, 0, 0, 625, 33554432, 0, 0)
+                    else
+                        return quest:progressEvent(314)
```

Comment:
For this line I also learned that you only use: quest:event(314)
When using progressEvent it overrides any other dialogue attempted with this NPC and can prevent any other interactions with this guy.

Reply:
Replaced.